src/js/msp.js

```diff
@@ -311,6 +314,13 @@ var MSP = {
         return bufferOut;
     },
     send_message: function (code, data, callback_sent, callback_msp, doCallbackOnError) {
+        if(CONFIGURATOR.virtualMode){
+            if(callback_msp instanceof Function){
```

Comment:
Curious why you are using `instanceof Function`?

Reply:
There are calls like this: 
`MSP.send_message(MSPCodes.MSP_FEATURE_CONFIG, false, false);`

Reply:
Maybe it should be independent of previous line `if (CONFIGURATOR.virtualMode) {` unless it fails or only specific for `virtual` mode.
`obj` is used to pass `callback_msp` later on with `MSP.callbacks.push(obj)`?

Reply:
I can probably replace it with a false check like in:
 `var obj = {'code': code, 'requestBuffer': bufferOut, 'callback': callback_msp ? callback_msp : false, 'timer': false, 'callbackOnError': doCallbackOnError};
`
But I'm not sure if there are any more special cases.

Reply:
`MSP.send_message` is the only one using this specific callback.

Reply:
thanks for checking it out